rust/src/fvm/machine.rs

```diff
@@ -147,8 +151,29 @@ pub unsafe extern "C" fn fil_create_fvm_machine(
         let blockstore = blockstore.finish();
 
         let externs = CgoExterns::new(externs_id);
+
+        let engine = match (|| -> anyhow::Result<Engine> {
+            let mut engines = ENGINES
+                .lock()
+                .map_err(|_| anyhow::Error::msg("engines lock is poisoned"))?;
```

Comment:
in general there is no need to catch this, you can just unwrap it, given this indicates a previous panic, and we already catch panics